### ISSUE TITLE
Issue with mixin() modifying returned object

### ISSUE THREAD

**Issue by rigikulm** [pinojs/pino#855]
I observed that mixin() incorrectly/unexpectedly modified the returned object which caused the wrong information to be shown in the log messages.

```javascript
const pino = require('pino');
const mixin = {
        app: 'My App'
};
const pinoOpt = {
     mixin() {
         return mixin; // This does not work as expected
    },
    timestamp: pino.stdTimeFunctions.isoTime
};

let log = pino(pinoOpt);
let info = {code: 0x112233,
                 description: 'Something bad happened'
};
  
log.info('Message 1');
log.warn(info, 'Warning message 1 with info object');
log.warn('Message 2');
```

Produces the following output:

```json
{
  "level": 30,
  "time": "2020-05-31T20:08:55.429Z",
  "pid": 23128,
  "hostname": "f01898963364",
  "app": "MyApp",
  "msg": "Message 1"
}
{
  "level": 40,
  "time": "2020-05-31T20:08:55.429Z",
  "pid": 23128,
  "hostname": "f01898963364",
  "app": "MyApp",
  "msg":"Warning message 1 with info \
     object",
  "code": 1122867,
  "descr": "Something happened"
}
{
  "level": 40,
  "time": "2020-05-31T20:08:55.429Z",
  "pid": 23128,
  "hostname": "f01898963364",
  "app": "MyApp",
  "msg": "Message 2",
  "code": 1122867,                                 <------ Should not be here
  "descr": "Something happened"       <------- Should not be here
}
```

If I refactor the mixin() function to instead return a new object instance each time, the code works as expected:

```javascript
const pinoOpt = {
     mixin() {
         return {app: 'MyApp'};       // This works as expected
    },
    timestamp: pino.stdTimeFunctions.isoTime
};
```

It seems like Pino should make a copy of the returned mixin object and then modify it as needed behind the scenes. The alternative would be to update the documentation for mixin() and explain that the object will be modified and that applications should make a copy themselves to avoid side effects.

**Comment by mcollina** [comment]
The feature currently work as expected. Note that using a shared object completely defeat the usage of `mixin()`. If you want to always add the same properties, _use a child logger_.

Adding some notes on the docs would be nice.

**Comment by davidmarkclements** [comment]
Closed by #857
